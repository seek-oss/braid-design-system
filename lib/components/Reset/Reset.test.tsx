import React, { AllHTMLAttributes } from 'react';
import { Reset } from '..';
import { render } from 'react-testing-library';
import forEachTheme from '../../test/utils/forEachTheme';

describe('Reset', () => {
  forEachTheme(({ ThemeProvider }) => {
    test('Known element', () => {
      const { container } = render(
        <ThemeProvider>
          <Reset component="section">Children</Reset>
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Unknown element', () => {
      const { container } = render(
        <ThemeProvider>
          <Reset component="custom-element">Children</Reset>
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Component', () => {
      function CustomComponent(props: AllHTMLAttributes<HTMLElement>) {
        return <span data-custom="component" {...props} />;
      }

      const { container } = render(
        <ThemeProvider>
          <Reset component={CustomComponent}>Children</Reset>
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
