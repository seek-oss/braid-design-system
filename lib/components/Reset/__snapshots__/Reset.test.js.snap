// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Theme: jobStreet Component 1`] = `
<span
  class=""
  data-custom="component"
>
  Children
</span>
`;

exports[`Reset Theme: jobStreet Known element 1`] = `
<section
  class="reset__section"
>
  Children
</section>
`;

exports[`Reset Theme: jobStreet Unknown element 1`] = `
<custom-element
  classname=""
>
  Children
</custom-element>
`;

exports[`Reset Theme: seekAnz Component 1`] = `
<span
  class=""
  data-custom="component"
>
  Children
</span>
`;

exports[`Reset Theme: seekAnz Known element 1`] = `
<section
  class="reset__section"
>
  Children
</section>
`;

exports[`Reset Theme: seekAnz Unknown element 1`] = `
<custom-element
  classname=""
>
  Children
</custom-element>
`;

exports[`Reset Theme: seekAsia Component 1`] = `
<span
  class=""
  data-custom="component"
>
  Children
</span>
`;

exports[`Reset Theme: seekAsia Known element 1`] = `
<section
  class="reset__section"
>
  Children
</section>
`;

exports[`Reset Theme: seekAsia Unknown element 1`] = `
<custom-element
  classname=""
>
  Children
</custom-element>
`;

exports[`Reset Theme: wireframe Component 1`] = `
<span
  class=""
  data-custom="component"
>
  Children
</span>
`;

exports[`Reset Theme: wireframe Known element 1`] = `
<section
  class="reset__section"
>
  Children
</section>
`;

exports[`Reset Theme: wireframe Unknown element 1`] = `
<custom-element
  classname=""
>
  Children
</custom-element>
`;
